motionplan/constraint.go

```diff
@@ -130,11 +130,14 @@ func (c *ConstraintHandler) CheckStateConstraintsAcrossSegment(ci *ik.Segment, r
 
 	for i := 0; i <= steps; i++ {
 		interp := float64(i) / float64(steps)
-		interpConfig := referenceframe.InterpolateInputs(ci.StartConfiguration, ci.EndConfiguration, interp)
-		interpC := &ik.State{Frame: ci.Frame, Configuration: interpConfig}
-		err = resolveStatesToPositions(interpC)
-		if err != nil {
-			return false, nil
+		interpC := &ik.State{Frame: ci.Frame}
+		if ci.StartPosition != nil && ci.EndPosition != nil && ci.StartConfiguration == nil || ci.EndConfiguration == nil {
```

Comment:
Can you comment what this is supposed to do?  As its written now it doesn't do what I think its supposed to be doing

Reply:
Sorry for the drive by comment (I'm good with whatever the team decides), but it seems like the caller needs to know that this branch exists in this function in order to know how to construct the provided segment.

IMO that is a pretty implicit API which is very easy to get wrong.  

I'm wondering if we might want to have this function:
1. Take an `interpralationMode` enum parameter which can be either  `configuration` or `pose`
2. Have this function return an error if the segment isn't properly formed to act on the specified mode i.e. if `pose` is provided but there is a missing start or end pose.

What do you think @biotinker?

Reply:
I was under the impression the conditional was well-formed https://go.dev/play/p/NHyWMU2mNtG

I will add an `interpralationMode` enum parameter which can be either configuration or pose

If we are using an enum param, we must check for if the segment is/isn't properly formed and also return an error form this function